classes/steps/lookups/cohort_lookup_step.php

```diff
@@ -0,0 +1,145 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_trigger\steps\lookups;
+
+defined('MOODLE_INTERNAL') || die;
+/**
+ * A lookup step that takes a user's ID and returns a string of
+ * all the cohorts that the user is currently assigned to
+ *
+ * @package    tool_trigger
+ * @author     Paul Damiani <Paul Damiani@catalyst-au.net>
+ * @copyright  Catalyst IT, 2018
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class cohort_lookup_step extends base_lookup_step
+{
+
+    use \tool_trigger\helper\datafield_manager;
+
+    /**
+     * The data field to get the cohort id from.
+     * @var string
+     */
+    private $cohortidfield = null;
+
+    /**
+     * The prefix to put before the new fields added to the workflow data.
+     *
+     * @var string
+     */
+    private $outputprefix = null;
+
+    /**
+     * The fields suplied by this step.
+     * A string containing all the cohorts a user is assigned to.
+     *
+     * @var array
+     */
+    private static $stepfields = array(
+        'id_[cohortid]'
+    );
+
+    protected function init()
+    {
+        $this->useridfield = $this->data['useridfield'];
+        $this->outputprefix = $this->data['outputprefix'];
+    }
+
+    /**
+     * {@inheritDoc}
+     * @see \tool_trigger\steps\base\base_step::execute()
+     */
+    public function execute($step, $trigger, $event, $stepresults)
+    {
+        global $DB;
+        $cohorts = "";
+        $datafields = $this->get_datafields($event, $stepresults);
+
+        if (!array_key_exists($this->useridfield, $datafields)) {
+            throw new \invalid_parameter_exception("Specified userid field not present in the workflow data: "
+                . $this->useridfield);
+        }
+
+        $sql = 'SELECT cohortid FROM {cohort_members} WHERE userid = :userid';
+        $params['userid'] = $datafields[$this->useridfield];
+        $cohortdata = $DB->get_recordset_sql($sql, $params);
+
+        foreach ($cohortdata as $key => $value) {
+            $stepresults[$this->outputprefix . 'id_' . $key] = $cohorts;
```

Comment:
$cohorts is an empty string, please set it to the name of the cohort.

Also don't use recordset's unless you really need to. By 'need to' I mean you have run into a production scaling issue and have profiling data to prove it. For small data sets they are actually slower than a normal get_records() as they go back to the database for every row instead of one chunk for the lot. I know this is copied code from the role step so not your fault but lets squash this habit before it forms.